src/core/openthread-core-default-config.h

```diff
@@ -912,7 +912,54 @@
 #define OPENTHREAD_CONFIG_ENABLE_STEERING_DATA_SET_OOB          0
 #endif
 
-/*
+/**
+ * @def OPENTHREAD_CONFIG_CHANNEL_MONITOR_SAMPLE_INTERVAL
+ *
+ * The sample interval in milliseconds used by Channel Monitoring feature.
+
+ * When enabled every sample interval a zero-duration Energy Scan is performed on every channel collecting a single
```

Comment:
nit: the sentence read strangely for me.  Suggestion: "When enabled, a zero-duration Energy Scan is performed, collecting a single RSSI sample per channel during each interval."